src/Compilers/CSharp/Test/Semantic/IOperation/IOperationTests_IIfStatement.cs

```diff
@@ -869,6 +869,340 @@ private static void A(bool flag, int number)
+    {
+    }
+
+    private void M(bool flag)
+    {
+        /*<bind>*/if (flag)
+        else
+        {
+            Op();
+        }
+/*</bind>*/    }
+}
+";
+            string expectedOperationTree = @"
+IConditionalOperation (OperationKind.Conditional, Type: null, IsInvalid) (Syntax: 'if (flag) ... }')
+  Condition: 
+    IParameterReferenceOperation: flag (OperationKind.ParameterReference, Type: System.Boolean) (Syntax: 'flag')
+  WhenTrue: 
+    IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '')
+      Expression: 
+        IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
+          Children(0)
+  WhenFalse: 
+    IBlockOperation (1 statements) (OperationKind.Block, Type: null) (Syntax: '{ ... }')
+      IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: 'Op();')
+        Expression: 
+          IInvocationOperation ( void P.Op()) (OperationKind.Invocation, Type: System.Void) (Syntax: 'Op()')
+            Instance Receiver: 
+              IInstanceReferenceOperation (ReferenceKind: ContainingTypeInstance) (OperationKind.InstanceReference, Type: P, IsImplicit) (Syntax: 'Op')
+            Arguments(0)
+";
+            var expectedDiagnostics = new DiagnosticDescription[] {
+                // file.cs(12,28): error CS1525: Invalid expression term 'else'
+                //         /*<bind>*/if (flag)
+                Diagnostic(ErrorCode.ERR_InvalidExprTerm, "").WithArguments("else").WithLocation(12, 28),
+                // file.cs(12,28): error CS1002: ; expected
+                //         /*<bind>*/if (flag)
+                Diagnostic(ErrorCode.ERR_SemicolonExpected, "").WithLocation(12, 28)
+            };
+
+            VerifyOperationTreeAndDiagnosticsForTest<IfStatementSyntax>(source, expectedOperationTree, expectedDiagnostics);
+        }
+
+        [CompilerTrait(CompilerFeature.IOperation)]
+        [Fact, WorkItem(27866, "https://github.com/dotnet/roslyn/issues/27866")]
+        public void IIfstatementWithIfKeywordMissingAndDoubleElseKeywordsPresent()
+        {
+            string source = @"
+using System;
+
+class P
+{
+    private void Op()
+    {
+    }
+
+    private void M()
+    {
+        /*<bind>*/{
+            else
+            {
+            }
+            else
+            {
+                Op();
+            }
+        }/*</bind>*/
+    }
+}
+";
+            string expectedOperationTree = @"
+IBlockOperation (2 statements) (OperationKind.Block, Type: null, IsInvalid) (Syntax: '{ ... }')
+  IConditionalOperation (OperationKind.Conditional, Type: null, IsInvalid) (Syntax: 'else ... }')
+    Condition: 
+      IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
+        Children(0)
+    WhenTrue: 
+      IExpressionStatementOperation (OperationKind.ExpressionStatement, Type: null) (Syntax: '')
+        Expression: 
+          IInvalidOperation (OperationKind.Invalid, Type: null) (Syntax: '')
```

Comment:
Can you find out what BoundNode this corresponds to? You can set a breakpoint in CSharpOperationFactory.CreateInternal and look in the debugger for the type. I'm not sure why we're getting an invalid node here, I was expected a block with 0 statements like WhenFalse.

Reply:
Trace of `boundNode.Kind`s in `CreateInternal(BoundNode boundNode)` for this particular UT:
```txt
Block
IfStatement
IfStatement
Conversion
BadExpression
ExpressionStatement
BadExpression
Block
Conversion
BadExpression
ExpressionStatement
BadExpression
Block
ExpressionStatement
Call
ThisReference
```

I conclude this is a `BadExpression` due to it being missing.